include/seqan3/io/sequence/sequence_file_out.hpp

```diff
@@ -0,0 +1,922 @@
+
+    static_assert([] () constexpr
+                  {
+                      for (field f : selected_field_ids::as_array)
+                          if (!field_ids::contains(f))
+                              return false;
+                      return true;
+                  }(),
+                  "You selected a field that is not valid for sequence files, please refer to the documentation "
+                  "of sequence_file_out::field_ids for the accepted values.");
+
+    static_assert([] () constexpr
+                  {
+                      return !(selected_field_ids::contains(field::SEQ_QUAL) &&
+                               (selected_field_ids::contains(field::SEQ) ||
+                               (selected_field_ids::contains(field::QUAL))));
+                  }(),
+                  "You may not select field::SEQ_QUAL and either of field::SEQ and field::QUAL at the same time.");
+
+    /*!\name Range associated types
+     * \brief Most of the range associated types are `void` for output ranges.
+     * \{
+     */
+    using value_type        = void;
+    using reference         = void;
+    using const_reference   = void;
+    using size_type         = void;
+    //!\brief A signed integer type, usually std::ptrdiff_t.
+    using difference_type   = std::ptrdiff_t;
+    //!\brief The iterator type of this view (an output iterator).
+    using iterator          = detail::out_file_iterator<sequence_file_out>;
+    //!\brief The const iterator type is void, because files are not const-iterable.
+    using const_iterator    = void;
+    //!\brief The type returned by end().
+    using sentinel          = ranges::default_sentinel;
+    //!\}
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    //!\brief Default constructor is explicitly deleted, you need to give a stream or file name.
+    sequence_file_out() = delete;
+    //!\brief Copy construction is explicitly deleted, because you can't have multiple access to the same file.
+    sequence_file_out(sequence_file_out const &) = delete;
+    //!\brief Copy assignment is explicitly deleted, because you can't have multiple access to the same file.
+    sequence_file_out & operator=(sequence_file_out const &) = delete;
+    //!\brief Move construction is defaulted.
+    sequence_file_out(sequence_file_out &&) = default;
+    //!\brief Move assignment is defaulted.
+    sequence_file_out & operator=(sequence_file_out &&) = default;
+    //!\brief Destructor is defaulted.
+    ~sequence_file_out() = default;
+
+    /*!\brief Construct from filename.
+     * \param[in] _file_name    Path to the file you wish to open.
+     * \param[in] fields_tag    A seqan3::fields tag. [optional]
+     *
+     * \details
+     *
+     * In addition to the file name, you may specify a custom seqan3::fields type which may be easier than
+     * defining all the template parameters.
+     */
+    sequence_file_out(filesystem::path const & _file_name,
```

Comment:
Sometimes you use leading underscores `(_file_name)`, sometimes not. Can we omit them or is there a meaning for that?

Reply:
I just use, because `file_name` is already used for the member variable.